api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -426,6 +445,100 @@ public void setHttpOnly(boolean isHttpOnly) {
      * @since Servlet 3.0
      */
     public boolean isHttpOnly() {
-        return isHttpOnly;
+        return Boolean.parseBoolean(getAttribute(HTTP_ONLY));
+    }
+
+    /**
+     * Sets the value of the cookie attribute associated with the given name.
+     * 
+     * <p>
+     * This should sync to any predefined attribute for which already a getter/setter pair exist in the current version,
+     * except for <code>version</code>. E.g. when <code>cookie.setAttribute("domain", domain)</code> is invoked, then
+     * <code>cookie.getDomain()</code> should return exactly that value, and vice versa.
+     * 
+     * @param name the name of the cookie attribute to set the value for, case insensitive
+     * 
+     * @param value the value of the cookie attribute associated with the given name, can be {@code null}
+     *
+     * @throws IllegalArgumentException if the cookie name is null or empty or contains any illegal characters (for example,
+     * a comma, space, or semicolon) or matches a token reserved for use by the cookie protocol
+     * 
+     * @since Servlet 5.1
+     */
+    public void setAttribute(String name, String value) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException(createErrorMessage("err.cookie_attribute_name_blank"));
+        }
+
+        if (hasReservedCharacters(name)) {
+            throw new IllegalArgumentException(createErrorMessage("err.cookie_attribute_name_is_token", name));
+        }
+
+        if (MAX_AGE.equalsIgnoreCase(name) && value != null) {
+            Long.parseLong(value);
```

Comment:
Use `Integer` to be consistent with `setMaxAge(int)` ?

Reply:
I now just call `setMaxAge` here, rather than duplicate the checking logic.